daemon/algod/api/server/v2/handlers.go

```diff
@@ -447,3 +450,23 @@ func (v2 *Handlers) GetPendingTransactions(ctx echo.Context, params generated.Ge
 func (v2 *Handlers) GetPendingTransactionsByAddress(ctx echo.Context, addr string, params generated.GetPendingTransactionsByAddressParams) error {
 	return v2.getPendingTransactions(ctx, params.Max, params.Format, &addr)
 }
+
+// Compile TEAL code to binary, return both binary and hash
+// (POST /v2/compile)
+func (v2 *Handlers) TealCompile(ctx echo.Context) error {
+	buf := new(bytes.Buffer)
```

Comment:
In other places we carefully have a limited reader from Request.Body; it'd be good to do that too here with a max source size of 100kB or so. See net/http MaxBytesReader which we use a couple times.

Reply:
added that in the new commit.